test/integration/trivial_test.rb

```diff
@@ -23,6 +23,40 @@ def test_copy_data_with_writes_to_source
     assert_test_table_is_identical
   end
 
+  class AlterTableDataWriter
+    include DbHelper
+
+    def initialize
+      @client = Mysql2::Client.new(source_db_config)
+    end
+
+    def start
+      table_name = full_table_name(DEFAULT_DB, DEFAULT_TABLE)
+      @client.query("INSERT INTO #{table_name} VALUES (9000, 'test')")
+      @client.query("ALTER TABLE #{table_name} ADD INDEX (data(100))")
+      @client.query("INSERT INTO #{table_name} (id, data) VALUES (9001, 'test')")
+    end
+  end
+
+  def test_copy_data_with_alter_fails_part_way_through
+    seed_simple_database_with_single_table
+
+    datawriter = AlterTableDataWriter.new
+    ghostferry = new_ghostferry(MINIMAL_GHOSTFERRY)
+
+    ghostferry.on_status(GhostferryHelper::Ghostferry::Status::BINLOG_STREAMING_STARTED) do
+      datawriter.start
```

Comment:
The `datawriter` concept is mostly for when you want a background thread hitting MySQL in the test.

In this case, I think you can just go directly to:

```suggestion
      source_db.query(....)
      source_db.query(....)
```

See types_test for more examples of this. This should save you from using another class.